### ISSUE TITLE
CrumbException not being honoured - CSRF issue

### ISSUE THREAD

**Issue by bugsduggan** [jenkinsci/ghprb-plugin#588]
I have setup a hook in GitHub pointing to `https://my-jenkins/ghprbhook` (where `my-jenkins` is the actual address of my jenkins setup).

When I test the hook, I can see jenkins returning `Invalid HTTP Response: 403` to GitHub.

I can also see in jenkins logs `No valid crumb was included in request for /ghprbhook. Returning 403.`

If I disable jenkins' CSRF protection, jenkins instead returns `Invalid HTTP Response: 302`, still not working but clearly hitting a different problem - I'll cross that bridge when I come to it.

All this suggests that the CrumbException is not being honoured when using this plugin.

dep  | version
------ | ------
Jenkins | 2.73.2
ghprb | 1.39.0
credentials | 2.1.16
matrix-project | 1.8
build-flow-plugin | 0.20 
ssh-agent | 1.15
structs | 1.10
github | 1.28.1
git | 3.6.3
github-api | 1.90
plain-credentials | 1.3
job-dsl | not installed
token-macro | 2.0

I'm happy to provide more config/context if needed.

Any help/pointers would be greatly appreciated.

**Comment by donnellyk** [comment]
Unrelated to your crumb problem, but I was googling for an issue and it led me here:

The solution to your `Invalid HTTP Response: 302` error is _probably_ not having a trailing slash on the web hook URL: `https://my-jenkins/ghprbhook/`

Hope that helps.

**Comment by bugsduggan** [comment]
@donnellyk you are correct, that was the issue that caused the `302`. Adding the slash fixed that problem.

I'm currently running with CSRF disabled globally on jenkins but obviously I'd like to not have to do that.

**Comment by brianjmurrell** [comment]
So is disabling CSRF globally what everyone using this plugin and webhooks really has to do?

**Comment by bjoernhaeuser** [comment]
We are running with CSRF and this plugin successfully.
